management-api-server/src/main/java/com/datastax/mgmtapi/Cli.java

```diff
@@ -436,18 +452,85 @@ void preflightChecks() {
     checkUnixSocket();
   }
 
-  private NettyJaxrsServer startHTTPService(String hostname, int port) throws SSLException {
+  @VisibleForTesting
+  void createSSLContext() throws SSLException {
+    this.sslContext =
+        SslContextBuilder.forServer(tlsCert, tlsKey)
+            .trustManager(tlsCaCert)
+            .clientAuth(ClientAuth.REQUIRE)
+            .ciphers(null, IdentityCipherSuiteFilter.INSTANCE)
+            .build();
+  }
+
+  @VisibleForTesting
+  void createSSLWatcher() throws IOException {
+    // Watch for tls_cert_file, tls_key_file and tls_ca_cert_file, add all their directories to
+    // Filesystem Watcher
+    WatchService watchService = FileSystems.getDefault().newWatchService();
+    java.nio.file.Path tlsCertParent = tlsCert.toPath().getParent();
+    java.nio.file.Path tlsKeyParent = tlsKey.toPath().getParent();
+    java.nio.file.Path tlsCaCertParent = tlsCaCert.toPath().getParent();
+
+    tlsCertParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+    tlsKeyParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+    tlsCaCertParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    executorService.execute(
+        () -> {
+          while (true) {
+            try {
+              WatchKey key = watchService.take();
+              List<WatchEvent<?>> events = key.pollEvents();
+              boolean reloadNeeded = false;
+              for (WatchEvent<?> event : events) {
+                WatchEvent.Kind<?> kind = event.kind();
+
+                WatchEvent<java.nio.file.Path> ev = (WatchEvent<java.nio.file.Path>) event;
+                java.nio.file.Path eventFilename = ev.context();
+
+                if (tlsCertParent.resolve(eventFilename).equals(tlsCert.toPath())
+                    || tlsKeyParent.resolve(eventFilename).equals(tlsKey.toPath())
+                    || tlsCaCertParent.resolve(eventFilename).equals(tlsCaCert)) {
+                  // Something in the TLS has been modified.. recreate SslContext
+                  reloadNeeded = true;
+                }
+              }
+              if (!key.reset()) {
+                // The watched directories have disappeared..
+                break;
+              }
+              if (reloadNeeded) {
+                createSSLContext();
+              }
+            } catch (InterruptedException e) {
```

Comment:
Suggestion: Won't it only be interrupted as part of application shut down? Maybe no need to log here.

Reply:
Threads can get interruptedexception without shutting down. It shouldn't happen, but if it does then that's an indication of something else.

Reply:
Can we distinguish between the two cases and only log in the case where the InterruptedException does not relate to application shutdown? I don't like the idea of producing errors in the logs for normal application behaviour.

Reply:
It is never normal application behavior to get InterruptedException.